src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,238 @@
+                // binary expression to reflect that it will never be negative.
+                // For example, the expression Array.Length > 0, becomes Array.Length == 0 when negated.
+                var operation = semanticModel.GetOperation(binaryExpression);
+
+                if (IsSpecialCaseBinaryExpression(operation as IBinaryOperation, cancellationToken))
+                {
+                    negatedOperatorType = binaryExpression.OperatorToken.Kind() == SyntaxKind.EqualsEqualsToken
+                        ? binaryExpression.Right is LiteralExpressionSyntax ? SyntaxKind.GreaterThanToken : SyntaxKind.LessThanToken
+                        : SyntaxKind.EqualsEqualsToken;
+                    negatedExpressionType = binaryExpression.Kind() == SyntaxKind.EqualsExpression
+                        ? binaryExpression.Right is LiteralExpressionSyntax ? SyntaxKind.GreaterThanExpression : SyntaxKind.LessThanExpression
+                        : SyntaxKind.EqualsExpression;
+                }
+
+                result = SyntaxFactory.BinaryExpression(
+                    negatedExpressionType,
+                    binaryExpression.Left,
+                    SyntaxFactory.Token(
+                        binaryExpression.OperatorToken.LeadingTrivia,
+                        negatedOperatorType,
+                        binaryExpression.OperatorToken.TrailingTrivia),
+                    binaryExpression.Right);
+
+                return true;
+            }
+
+            result = null;
+            return false;
+        }
+
+        private ExpressionSyntax Negate(ExpressionSyntax expression, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            if (TryNegateBinaryComparisonExpression(expression, semanticModel, cancellationToken, out var result))
+            {
+                return result;
+            }
+
+            switch (expression.Kind())
+            {
+                case SyntaxKind.ParenthesizedExpression:
+                {
+                    var parenthesizedExpression = (ParenthesizedExpressionSyntax)expression;
+                    return parenthesizedExpression
+                        .WithExpression(Negate(parenthesizedExpression.Expression, semanticModel, cancellationToken))
+                        .WithAdditionalAnnotations(Simplifier.Annotation);
+                }
+
+                case SyntaxKind.LogicalNotExpression:
+                {
+                    var logicalNotExpression = (PrefixUnaryExpressionSyntax)expression;
+
+                    var notToken = logicalNotExpression.OperatorToken;
+                    var nextToken = logicalNotExpression.Operand.GetFirstToken(
+                        includeZeroWidth: true, includeSkipped: true, includeDirectives: true, includeDocumentationComments: true);
+
+                    var existingTrivia = SyntaxFactory.TriviaList(
```

Comment:
i think this needs comments explaining what this is for.

Reply:
@CyrusNajmabadi , upon closer inspection I'm not sure this is needed.  I added a test case for `if /*0*/(/*1*/!a/*2*/)/*3*/ { a(); } else { b(); }`  and  notToken doesn't have any leading or trailing trivia.  Is there any way you know of for the not Token to have trivia?

Reply:
sure.    ```if (!/**/a)```

Reply:
Ah, that's the one.  Putting trivia between the not Token and the identifier is the test case needed here.  Will add the comment and the test case.  Thanks!